src/main/host/syscall/poll.c

```diff
@@ -220,13 +220,18 @@ SysCallReturn syscallhandler_ppoll(SysCallHandler* sys, const SysCallArgs* args)
         return (SysCallReturn){.state = SYSCALL_DONE, .retval.as_i64 = result};
     }
 
-    const struct timespec* ts_timeout;
+    struct timespec ts_timeout_val;
+    struct timespec* ts_timeout = NULL;
 
     if (!ts_timeout_ptr.val) {
         // NULL timespec is valid and means infinite timeout
         ts_timeout = NULL;
     } else {
-        ts_timeout = process_getReadablePtr(sys->process, ts_timeout_ptr, sizeof(*ts_timeout));
+        if (process_readPtr(
+                sys->process, &ts_timeout_val, ts_timeout_ptr, sizeof(ts_timeout_val)) != 0) {
+            return (SysCallReturn){.state = SYSCALL_DONE, .retval.as_i64 = -EFAULT};
+        }
+        ts_timeout = &ts_timeout_val;
```

Comment:
Can we remove the second timespec `ts_timeout` here?

Reply:
Sort of - I got rid of the local pointer and just did the "pointer selection" in the call to `_syscallhandler_pollHelper`. Commented and rearranged a bit to hopefully clarify what's going on here.